cmd/sonobuoy/app/args.go

```diff
@@ -168,62 +172,48 @@ const (
 // users. Using e2e-parallel incorrectly has the potential to destroy clusters!
 func AddE2EConfigFlags(flags *pflag.FlagSet) *pflag.FlagSet {
 	e2eFlags := pflag.NewFlagSet("e2e", pflag.ExitOnError)
-	modeName := ops.NonDisruptiveConformance
-	defaultMode := modeName.Get()
-	e2eFlags.String(
-		e2eFocusFlag, defaultMode.E2EConfig.Focus,
-		"Specify the E2E_FOCUS flag to the conformance tests. Overrides --mode.",
-	)
-	e2eFlags.String(
-		e2eSkipFlag, defaultMode.E2EConfig.Skip,
-		"Specify the E2E_SKIP flag to the conformance tests. Overrides --mode.",
-	)
-	e2eFlags.String(
-		e2eParallelFlag, defaultMode.E2EConfig.Parallel,
-		"Specify the E2E_PARALLEL flag to the conformance tests. Overrides --mode.",
```

Comment:
Again a bit of a simplification here that I dont think loses anything. We had to nest flags within our types so we could do checking later if the flag was provided or a default value. Now we just say: let the flags figure it out.

If they provide the flags in a conflicting way (--mode=quick --e2e-focus=foo) then the flags will sort of fight over the value; last to set it wins.

I think this is reasonable and I dont see a need to bend over backwards to support wonkily-written commands like that.

Reply:
Agreed.
I would only suggest that if both `mode` and `e2e-focus` flags are provided together to, maybe, output a warning.

Reply:
OK; added a warning in a persistentPrerun method now. Otherwise we have to place it in a bunch of different places. Its a simple check and would apply anywhere those flags are present/provided.